### DISCUSSION TITLE
为什么不把定义可变变量改为mut，非要在前面加一个let，程序员懒不想写这么多，并且只写一个mut看着也舒服

### DISCUSSION THREAD

**Discussion by tomener** [moonbitlang/core#238] [Ideas]
`
//定义可变变量
let a = 100

//定义可变变量
mut b = 200
`

**Comment by Yoorkin** [comment]
单独写mut允许对绑定有更精细的可变性控制。MoonBit未来会有类似的东西：

```
let (mut a, b) = get_two() // a可变，b不可变
```

这样一来let的语法是`let <pattern> = <expr>`，`mut`是pattern的一部分。这个`struct`里的`mut`语法是一致的。

更多的例子：

```
match foo() {
    Some(mut x) => ... 
    None => ...
}
```

```
fn {
    Some({ field1, mut field2 }) => ... // field1不可变，field2可变
    None => ...
}
```

**Comment by Lampese** [comment]
> 单独写mut允许对绑定有更精细的可变性控制。MoonBit未来会有类似的东西：
> 
> ```
> let (mut a, b) = get_two() // a可变，b不可变
> ```
> 
> 这样一来let的语法是`let <pattern> = <expr>`，`mut`是pattern的一部分。这个`struct`里的`mut`语法是一致的。
> 
> 更多的例子：
> 
> ```
> match foo() {
>     Some(mut x) => ... 
>     None => ...
> }
> ```
> 
> ```
> fn {
>     Some({ field1, mut field2 }) => ... // field1不可变，field2可变
>     None => ...
> }
> ```

所以不用 var 是要考虑以后结构语法里面有 mut 的语义一致性吗？

**Reply by Yoorkin** [reply]
是为了方便标注哪些binder是可变的，其次是语法风格更一致
